contracts/tokenomics/generator/src/contract.rs

```diff
@@ -1326,6 +1327,82 @@ fn query_orphan_proxy_rewards(deps: Deps, lp_token: Addr) -> Result<Uint128, Con
     Ok(pool.orphan_proxy_rewards)
 }
 
+/// ## Description
+/// Returns an [`ContractError`] on failure, otherwise returns information about the LP Pool rewards
+/// configs in a [`PoolInfoResponse`] object .
+/// ## Params
+/// * **deps** is the object of type [`Deps`].
+/// * **env** is the object of type [`Env`].
+/// * **lp_token** is the object of type [`Addr`].
+fn query_pool_info(
+    deps: Deps,
+    env: Env,
+    lp_token: Addr,
+) -> Result<PoolInfoResponse, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    let lp_token = addr_validate_to_lower(deps.api, lp_token.as_str())?;
+    let pool = POOL_INFO.load(deps.storage, &lp_token)?;
+
+    let lp_supply: Uint128;
+    let mut pending_on_proxy = None;
+    let mut pending_astro_rewards = Uint128::zero();
+
+    // If proxy rewards are live for this LP token, calculate its Pending Proxy rewards
+    match &pool.reward_proxy {
+        Some(proxy) => {
+            lp_supply = deps
+                .querier
+                .query_wasm_smart(proxy, &ProxyQueryMsg::Deposit {})?;
+
+            // If LP tokens are staked via proxy contract, fetch currently pending proxy rewards
+            if !lp_supply.is_zero() {
+                let res: Option<Uint128> = deps
+                    .querier
+                    .query_wasm_smart(proxy, &ProxyQueryMsg::PendingToken {})?;
+
+                if let Some(token_rewards) = res {
```

Comment:
"res" will always represent the some value. Maybe use something like this: 
if res > 0 { 
    pending_on_proxy = Some(res);
}